### DISCUSSION TITLE
PHP 8.1 - Smarty 4.1 Deprecated preg_replace(): Passing null to parameter #3 ($subject)

### DISCUSSION THREAD

**Discussion by cristianocaruso** [smarty-php/smarty#738] [Show and tell]
Hi,
my script gets me this error with preg_replace 

_`Deprecated: preg_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated in Smarty4\sysplugins\smarty_resource_recompiled.php(56) : eval()'d code on line 20`_

Searching in source I've found an issue with PHP 8.1 when parameter {$params[0]} is null so I used Elvis operator ( ?: ) to bypass null value

**_smartylibs/plugins/modifiercompiler.strip_tags.php_**

```
function smarty_modifiercompiler_strip_tags($params)
{
    if (!isset($params[ 1 ]) || $params[ 1 ] === true || trim($params[ 1 ], '"') === 'true') {
        return "preg_replace('!<[^>]*?>!', ' ', {$params[0]} ?: '')";
    } else {
        return 'strip_tags(' . $params[ 0 ] . ')';
    }
}

```

Maybe this helps someone, can you add to source?

**Comment by wisskid** [comment]
Ported in into https://github.com/smarty-php/smarty/pull/803
